config/deprecation.go

```diff
@@ -0,0 +1,359 @@
+package config
+
+import (
+	"fmt"
+	"log" //nolint:revive // log is ok here as the logging facility is not set-up yet
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/fatih/color"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/aggregators"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+// Escalation level for the plugin or option
+type Escalation int
+
+func (e Escalation) String() string {
+	switch e {
+	case Warn:
+		return "WARN"
+	case Error:
+		return "ERROR"
+	}
+	return "NONE"
+}
+
+const (
+	// None means no deprecation
+	None Escalation = iota
+	// Warn means deprecated but still within the grace period
+	Warn
+	// Error means deprecated and beyond grace period
+	Error
+)
+
+// deprecationInfo contains all important information to describe a deprecated entity
+type deprecationInfo struct {
+	// Name of the plugin or plugin option
+	Name string
+	// Level of deprecation
+	Level Escalation
```

Comment:
Can we rename this to `LogLevel` to be a bit more clear?

Reply:
In theory we could have more deprecation levels that not directly correspond to logging levels, but I guess that is currently not relevant. Will change...

Reply:
Fixed in the latest commits.